File: 0075-sort-colors/0075-sort-colors.cpp
class Solution {
public:
    void sortColors(vector<int>& nums) {
        int low,mid=0;
        int high=nums.size()-1;
        while(mid<=high){
            if(nums[mid]==0){
                swap(nums[mid],nums[low]);
                low++;
                mid++;
            }
            else if(nums[mid]==2){
                swap(nums[mid],nums[high]);
                high--;
            }
            else{
                mid++;
            }
        }
    }
};

File: 0075-sort-colors/0075-sort-colors.java
class Solution {
    //Dutch National Flag Approach
    public void sortColors(int[] nums) {
        Sort2(nums);
        }
     public static void Sort2(int[] nums){
        int low=0,mid=0,high=nums.length-1;
        while(mid<=high){
            if(nums[mid]==0) {Swap(nums,mid,low); low++;mid++;}
            else if((nums[mid]==2)){Swap(nums, mid, high);high--;}
            else {mid++;}
        }
    }
    public static void Swap(int [] nums,int a, int b){
        int temp=nums[a];
        nums[a]=nums[b];
        nums[b]=temp;
    }
}